### PR TITLE
testng_hw

### PR DESCRIPTION
**Pull request by kislayao**

### CODE REVIEW

File: src/test/java/CalculatorTest_Div_Int.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Test(dependsOnGroups = "smoke")
+    public void div_negative_number(){
+        int expectedValue = -2;
+        int actualValue = calculator.div(-4,2);
+        Assert.assertEquals(actualValue, expectedValue, "Values don't match");
+    }
+
+    @Test (groups = "smoke", dataProvider = "Data for div INT test", dataProviderClass = StaticProvider.class, threadPoolSize = 4)
```
Comment (wildpioneer) [inline src/test/java/CalculatorTest_Div_Int.java:47]: Отсутствуют тесты на:
1. @Parameters
2. retryAnalyzer
3. Double.NaN
4. Double.NEGATIVE_INFINITY
5. Double.POSITIVE_INFINITY

**Review by wildpioneer — CHANGES_REQUESTED** [review]

File: src/test/java/CalculatorTest_Div_Int.java (hunk 1 above)
Comment (kislayao) [reply src/test/java/CalculatorTest_Div_Int.java:47]: Double.POSITIVE_INFINITY test was in the CalculatorTest_Div_Double class
All other tests were added also to the same class

Also I've question:
I've tried to run div_test_INT.xml which has tests for CalculatorTest_Div_Int class and all the test failed with NullPointerException
But when I run tests in the CalculatorTest_Div_Int class itself everything is OK
It looks like calculator object becomes unreachable when I run test via div_test_INT.xml
Could you give me a hint what can be wrong?

Comment (wildpioneer) [reply src/test/java/CalculatorTest_Div_Int.java:47]: To find out what the issue with NPE - you have to review TestNG lesson. I explained the usage of Before/After annotations.

**Comment by wildpioneer** [comment]
Проверено

**Pull request closed without merging**